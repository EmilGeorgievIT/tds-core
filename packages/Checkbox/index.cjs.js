require('./dist/index.css')
const Checkbox = require('./dist/index.cjs')

module.exports = Checkbox
