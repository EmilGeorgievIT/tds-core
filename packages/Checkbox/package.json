{
  "name": "@tds/core-checkbox",
  "version": "1.1.11",
  "main": "index.cjs.js",
  "module": "index.es.js",
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/telusdigital/tds.git"
  },
  "publishConfig": {
    "access": "public"
  },
  "author": "TELUS digital",
  "engines": {
    "node": ">=8"
  },
  "bugs": {
    "url": "https://github.com/telusdigital/tds/issues"
  },
  "homepage": "http://tds.telus.com",
  "peerDependencies": {
    "react": ">=15",
    "react-dom": ">=15"
  },
  "dependencies": {
    "@tds/core-box": "^1.1.0",
    "@tds/core-decorative-icon": "^1.1.1",
    "@tds/core-input-feedback": "^1.0.8",
    "@tds/core-paragraph": "^1.0.8",
    "@tds/core-text": "^1.0.8",
    "prop-types": "^15.5.10"
  },
  "devDependencies": {
    "@tds/core-colours": "^1.1.1",
    "@tds/shared-choice": "^1.1.8"
  }
}
