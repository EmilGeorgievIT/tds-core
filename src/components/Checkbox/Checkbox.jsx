import React from 'react'
import PropTypes from 'prop-types'

import safeRest from '../../utils/safeRest'
import generateId from '../../utils/generateId'

import Text from '../Typography/Text/Text'

// import styles from './Checkbox.modules.scss'

const Checkbox = ({ label, ...rest }) => {
  const checkboxId = generateId(rest.id, rest.name, label)

  return (
    <label data-no-global-styles htmlFor={checkboxId.identity()}>
      <input {...safeRest(rest)} type="checkbox" data-no-global-styles id={checkboxId.identity()} />
      <Text size="medium">{label}</Text>
    </label>
  )
}

Checkbox.propTypes = {
  label: PropTypes.string.isRequired,
}

Checkbox.defaultProps = {}

export default Checkbox
