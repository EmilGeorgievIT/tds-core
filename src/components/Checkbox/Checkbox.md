```
<Checkbox label="Choice A one two three four five six seven eight nine ten eleven twelve" />
```
