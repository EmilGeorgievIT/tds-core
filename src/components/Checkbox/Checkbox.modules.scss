@import '../../scss/settings/variables';
@import '../../scss/utility/mixins';

/* FIXME: resetting globally scoped styles in form.scss
   because of input:not(type='radio')
*/
input[data-no-global-styles] {
  border: none;
  border-radius: 0;
  box-shadow: none;
  color: transparent;
  font-size: inherit;
  line-height: inherit;
  outline: inherit;
  padding: 0;
  width: auto;
  transition: none;
  font-family: inherit;

  &:focus {
    box-shadow: none;
    border-color: inherit;
  }

  @include from-breakpoint(medium) {
    letter-spacing: inherit;
  }
}
