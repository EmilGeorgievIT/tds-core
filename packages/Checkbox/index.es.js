import './dist/index.css'
import Checkbox from './dist/index.es'

export default Checkbox
