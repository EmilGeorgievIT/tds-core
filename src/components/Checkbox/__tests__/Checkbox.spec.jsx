import React from 'react'
import { shallow } from 'enzyme'

import Checkbox from '../Checkbox'

describe('Checkbox', () => {
  const defaultProps = {
    label: 'The input',
  }
  const doShallow = (overrides = {}) => shallow(<Checkbox {...defaultProps} {...overrides} />)
  const findInputElement = input => input.find('input')

  it('renders', () => {
    const checkbox = doShallow()

    expect(checkbox).toMatchSnapshot()
  })

  it('does other things', () => {
    const checkbox = doShallow()

    expect(checkbox).toBePresent()
  })

  it('passes additional attributes to the element', () => {
    const checkbox = doShallow({
      disabled: 'true',
      'data-some-attr': 'some value',
    })
    expect(findInputElement(checkbox)).toHaveProp('disabled', 'true')
    expect(findInputElement(checkbox)).toHaveProp('data-some-attr', 'some value')
  })

  it('does not allow custom CSS', () => {
    const checkbox = doShallow({
      className: 'my-custom-class',
      style: { color: 'hotpink' },
    })

    expect(findInputElement(checkbox)).not.toHaveProp('className', 'my-custom-class')
    expect(findInputElement(checkbox)).not.toHaveProp('style')
  })
})
