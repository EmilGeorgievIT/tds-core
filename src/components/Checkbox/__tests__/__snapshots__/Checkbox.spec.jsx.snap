// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox renders 1`] = `
<label
  data-no-global-styles={true}
  htmlFor="the-input"
>
  <input
    data-no-global-styles={true}
    id="the-input"
    type="checkbox"
  />
  <Text
    block={false}
    bold={false}
    invert={false}
    size="medium"
  >
    The input
  </Text>
</label>
`;
